src/mango/test/mango.py

```diff
@@ -156,13 +157,29 @@ def create_index(
         assert r.json()["name"] is not None
 
         created = r.json()["result"] == "created"
-        if created:
-            # wait until the database reports the index as available
-            while len(self.get_index(r.json()["id"], r.json()["name"])) < 1:
-                delay(t=0.1)
+        if created and wait_for_built_index:
+            # wait until the database reports the index as available and build
+            while (
+                len(
+                    [
+                        i
+                        for i in self.get_index(r.json()["id"], r.json()["name"])
+                        if i["build_status"] == "ready"
+                    ]
+                )
+                < 1
+            ):
+                delay(t=0.2)
 
         return created
 
+    def wait_for_built_indexes(self):
+        indexes = self.list_indexes()
+        while len(
+            [i for i in self.list_indexes() if i["build_status"] == "ready"]
+        ) < len(indexes):
+            delay(t=0.2)
```

Comment:
How about:

```python
all_built = True
while not all_built:
    for idx in self.list_indexes():
        if idx["build_status"] != "ready":
            all_built = False
            break
    if not all_built:
        delay(t = 0.2)
```

Reply:
This is the formatting python-black recommended

Reply:
Properly formatted weird Python is still weird Python.

Reply:
Another alternative that came to mind:

```python
while True:
    if all(idx["build_status"] == "ready" for idx in self.list_indexes()):
        break
    delay(t = 0.2)
```